.github/workflows/Tests.yaml

```diff
@@ -403,7 +402,7 @@ jobs:
           -D USE_CCACHE=ON
           -D BUILD_PYTHON_BINDINGS=${BUILD_PYTHON_BINDINGS:-'ON'}
           -D PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE:-'/usr/bin/python3'}
-          -D BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS:-'OFF'}
+          -D BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS:-'ON'}
```

Comment:
I don't yet understand how these shared-libs builds link in Charm++, since we don't even find the shared libs in `FindCharm`. Also your CI build https://github.com/sxs-collaboration/spectre/pull/2669/checks?check_run_id=1567661216 only finds the static libs:

```
-- CMAKE_CXX_LINK_FLAGS:  -L/work/charm_6_10_2/multicore-linux-x86_64-clang/lib -fuse-ld=gold
```
(Note the directory is `lib` not `lib_so`)

Can you illuminate me?

When I build with shared libs on macOS this is the issue:

```
$ make ErrorHandling
...
Linking CXX shared library ../../lib/libErrorHandling.dylib
Undefined symbols for architecture x86_64:
  "_CmiError", referenced from:
      auto void Parallel::printf_error<>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::'lambda'()::operator()<char const*>() const in AbortWithErrorMessage.cpp.o
  "_CmiMyPe", referenced from:
      Converse::CkMyPe() in AbortWithErrorMessage.cpp.o
  "_CmiPrintf", referenced from:
      Parallel::abort(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in AbortWithErrorMessage.cpp.o
      Parallel::abort(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in FloatingPointExceptions.cpp.o
  "_realCkExit", referenced from:
      Parallel::abort(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in AbortWithErrorMessage.cpp.o
      Parallel::abort(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in FloatingPointExceptions.cpp.o
ld: symbol(s) not found for architecture x86_64
```

Possibly related: `ErrorHandling` doesn't currently depend on `Parallel` but should, however `Parallel` depends on `ErrorHandling`, so this is a cyclic dependency.

Reply:
My understanding is that on Linux you can also stick static libraries into shared libraries on Linux as long as the static libraries are marked as position independent. I have no idea what macOS does.